src/process_utility.c

```diff
@@ -678,85 +682,134 @@ process_alter_column_type(Hypertable *ht, AlterTableCmd *cmd)
 }
 
 static void
-process_altertable(Node *parsetree)
+process_altertable_start(Node *parsetree)
 {
 	AlterTableStmt *stmt = (AlterTableStmt *) parsetree;
 	Oid			relid = AlterTableLookupRelation(stmt, NoLock);
-	Cache	   *hcache = NULL;
-	ListCell   *lc;
-	Hypertable *ht;
 
-	if (!OidIsValid(relid))
+	if (!OidIsValid(relid) || is_hypertable(relid))
```

Comment:
`is_hypertable()` checks if oid is valid so you can remove that check here.

Reply:
Won't work in this case since if `is_hypertable()` returns false, we cannot know if it was because this was a regular table or the OID wan't invalid. In the former case, we don't want to exit, in the latter case we want to exit.